leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -39,17 +38,526 @@ var EvolvingFunctionBBM = Backbone.Model.extend({
        
     },
     
-    //TODO: I think could have some helper functions for the constructor. 
+    //TODO: I think could have some helper functions for the constructor.
+    
+    // TODO update all of the function descriptions
+
+    /**
+     * Returns the 4 digit representation for this
+     * EvolvingFunction's last function segment's
+     * marked value value
+     */
+    getLastMarkedVal: function() {
+        var len = this.functionSegList.length;
+        if (len > 0) {
+            return this.functionSegList[len - 1].get('refEvidencePair');
+        }
+        // Original Function
+        // if (len > 0) {
+        //     return this.functionSegList[len - 1].funcX;
+        // }
+    },
+
+    /**
+     * Returns the 4 digit representation for this
+     * EvolvingFunction's second last function segment's
+     * marked value value. If there is no second last function
+     * segment, this function returns the 4 digit representation of 
+     * the only function segment's marked value
+     */
+    getSecondLastMarkedVal: function() {
+        var len = this.functionSegList.length;
+        if (len > 1) {
+            return this.functionSegList[len - 2].get('refEvidencePair');
+        } else {
+            return this.getLastMarkedVal();
+        }
+        // Original Function
+        // if (len > 1) {
+        //     return this.functionSegList[len - 2].funcX;
+        // } else {
+        //     return this.getLastMarkedVal();
+        // }
+    },    
+
+    /**
+     * Creates a new RepFuncSegment object containing function
+     * segments in the relative time interval [time1, time2], and add it to this
+     * EvolvingFunction's functionSegList, in place of the function
+     * segements in the time interval [time1, time2]
+     *
+     * @param {String} time1
+     *   first relative time point
+     * @param {String} time2
+     *   second relative time point
+     */
+    // TODO fix this function
+    // I think we should set it so that the repeating function starts at time 1 and ends at 
+    // time 2
+    setRepeatingFunction: function(time1, time2) {
```

Comment:
Isn't repAbsTime an input box in the frontend? Can you get the repAbsTime from there? Or does it make more sense to remove the input box if repAbsTime is just predetermined from time1 and time2? Also - is repAbsTime used for anything in this function? ALSO is time1 and time2 are strings how are we subtracting them? Lots of questions sorry!

Reply:
repAbsTime =  parseInt(time2) - parseInt(time1);

Reply:
Will probably have to discuss this with @i-m-foster